### ISSUE TITLE
lang attribute on <html> is set to a known language

### ISSUE THREAD

**Issue by jackdomleo7** [jackdomleo7/Checka11y.css#95]
### Describe the new a11y feature or project enhancement

It would be nice to indicate if the value set on the `lang` attribute on the `<HTML>` is set to a know language string as [defined here by w3](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry).
Although this is a very long piece of work and may prove to be better suited in a JS accessibility tool.

### Link(s)

- https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry
- https://www.w3.org/International/articles/language-tags/#rfc
- https://r12a.github.io/app-subtags

**Comment by SavvasStephanides** [comment]
Good idea. We could try using the [`@each` rule](https://sass-lang.com/documentation/at-rules/control/each) in Sass

**Comment by jackdomleo7** [comment]
I had an idea on this:
- Find an API that can provide us a JSON array of all the accepted `lang` codes _(struggling with this bit)_
- Create a JS script that will dynamically create a `_lang.scss` file with an SCSS equivalent array _(this'll be interestingly tricky)_
- Run this script on pre-build so it runs just before we convert the SCSS to CSS
- Voila!

**Comment by jackdomleo7** [comment]
I was able to find an API to do this. But after reading the w3 docs on the HTML `lang` attribute, it seems that a `lang` attribute can actually be very complex: [read here](https://www.w3.org/International/articles/language-tags/#rfc).

![image](https://user-images.githubusercontent.com/43371432/136626208-55411a1e-f1a0-4a34-b9f8-c21dc5c74694.png)

At the very most, I have only ever seen `language-region` (e.g. `en-GB`, `en-US` or `fr-CA`), but this isn't to say more complex `lang` tags exist out there.

I could easily check for just `language` (e.g. `en`) or go a step further and check for `language-region` and show a warning if it doesn't match, but this could give a false warning (especially if a webpage has used more attributes than just `language` and `region`). I feel trying to string all the `langiages`, `regions`, `extlangs`, `scripts`, etc would be a huge waste of time and may not be as a helpful feature as I originally anticipated.

Interested to hear your opinion @SavvasStephanides 🤔

**Comment by jackdomleo7** [comment]
Closing this issue since the `lang` attribute can have very complex structures. We already have an error that highlights if no `lang` is set on the `<HTML>` element, and this should be enough. We _could_ update the `codes.md` file for the `lang` error to encourage a correct format.
